### ISSUE TITLE
Reports - Meetings - Calendar Filter

### ISSUE THREAD

**Issue by JoeNewUser** [salesagility/SuiteCRM#2189]
I am trying to create a meeting report that shows the activities that are scheduled for various date ranges. 
#### Issue

These are the  5 reports that I tried to create and a brief summary of the problems that I encountered.  A detailed description of the problems are shown in the attached docx file.
1. A report showing all of the meetings.
   The report view and the meetings list view have a different meeting count: 280 vs 286.
   Adding the “Contact” data field to the report now shows 286 meetings.
2. A report showing the meetings on a given date.
   The date value changes when clicking on the “Update” button.
   No meetings in the resulting report when I know there are meetings on that date.
3. A report showing the meeting greater than or equal a  given date.
   Date value changes when clicking on the “Update” button.
   Clicking the “Next Screen Arrow” to view the next screen returns all of the meetings.  The filter is lost.
   Clicking the “Update” button again to refresh the report, decrements the date again.
   The data Export to a csv file results in all meetings, the filter condition is lost. 
4. A report showing the meetings less than or equal a given date.
   Again, the condition date changes when clicking “Update”.
   Export fails, only column headers in the csv file, even though there are 115 meetings in the report.
   Click next screen arrow to view the next group of meeting results in no meetings being displayed. Meetings went from 1 of 115, to 0 of 0.
5. A report showing the meetings between 2 dates
   Again, the condition dates changes when clicking “Update”.  However, both dates decrement which may or may not be the correct behavior depending upon the logic.  A screen with 1-20 of 99 meetings is displayed.
   Click the “next” screen arrow to view the next group of meeting results in no meetings being displayed.
   Export fails, only column headers in the csv file, even though there are 99 meetings in the report.
### Expected Behavior

Described in the attached docx file.

[SuiteCRM_MeetingReportProblems_06.docx](https://github.com/salesagility/SuiteCRM/files/465145/SuiteCRM_MeetingReportProblems_06.docx)
#### Actual Behavior

Described in the attached docx file.
#### Steps to Reproduce

See the attached docx file.
#### Context

I am not able to create reports showing meetings for various calendar time periods.
I think this should be a medium to high priority issue since multiple bugs may be involved.
#### Your Environment

See the attached docx file.

**Comment by horus68** [comment]
Your doc says you are using SuiteCRM 7.6.6.
Can you replicate using the more recent version 7.7.4?

**Comment by JoeNewUser** [comment]
Horus68, thank you for your assistance.

I did not want to upgrade to 7.7.4 on my laptop yet due to the extensive comments in the forum about problems.  If the report problems were resolved on the 7.7.4 demo, then I will look at upgrading the laptop.

So, I did a quick report check on the demo site which is at 7.7.4.

I created a report on the demo site with various "conditions".  The data fields were Start Date, Subject, Duration, and Assigned To.  Dates were selected by clicking on the calendar date in the test condition.

No Condition, saw 200 meetings.  The earliest was 8-31-16 and the latest was 8-31-17.  Export and PDF Export seemed to be OK.

Date Equal: 8-31-16, nothing, should be 1 meeting.  9-29-16, nothing, should be 2 meetings.

Date Greater than or Equal to 10-1-16.  188 meetings, Export OK, but clicking the next screen arrow showed 200 meetings.  I did not think to try the PDF Export before logging off the demo, sorry.

Dates between 9-1-16 and 9-30-16, nothing.

Dates Less Than or Equal 10-31-16, 26 meetings, Export fails, PDF fails, next screen fails (0 meetings).

Net-Net, I think version 7.7.4 still has the reporting condition issues.

There are some new test condition options, but not all were applicable: e.g. a specific date equal to was no listed.

The PDF Export tab was nice, but it should include column headers, datetime stamp, and page numbers.

Hope this helps.

**Comment by JoeNewUser** [comment]
My error in hitting the wrong button.  Sorry

**Comment by JoeNewUser** [comment]
I did some more testing tonight.

The date value is not decrementing on the demo version (7.7.4).  I forgot to add that comment in my previous posting.

I am using the existing meetings in the demo database, and I created my report.

Testing for meetings occurring on dates greater or equal to Oct 1st, 2016.  Seems to be OK.  I checked the PDF Export this time and it is working.

Testing for meetings between Sept 1st and Sept 30th.  Found 10 meetings, Export and PDF Export both fail.  But, this time it did find meetings.

Testing for meetings between Sept 1st and Oct 31st.  Found 25 meetings.  Export and PDF Export both fail.  The screen showed 1 to 20 of 25.  Clicked the "next" arrow to view the next set of meetings.  Nothing now 0 of 0.
